test/unit/services/AuthServicesTest.ts

```
@@ -0,0 +1,79 @@
+};
+
+const registerRequest: LoginRequest = {
+  email: "Test",
+  password: "Terry",
+};
+
+describe("AuthService", function () {
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe("login", function () {
+    it("should return token", async () => {
+      const data = [
+        "eyJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE3MzYzNDEyOTAsImV4cCI6MTczNjM3MDA5MCwiZW1haWwiOiJUZXJyeUBnbWFpbC5jb20iLCJzdWIiOiJUZXJyeUBnbWFpbC5jb20iLCJpc3MiOiJLYWlub3MifQ.Dv1_YV2zS_I09SFQ-Ybach1V_6nLxJr3uLkEmF3TaHk",
+      ];
+
+      mock
+        .onPost("http://localhost:8080/api/auth/login", loginRequest)
+        .reply(200, data);
+      const results = await getToken(loginRequest);
+
+      expect(results).to.deep.equal(data);
+    });
+
+    it("should throw exception when 500 returned by axios", async () => {
+      mock
+        .onPost("http://localhost:8080/api/auth/login", badLoginRequest)
+        .reply(500);
+      try {
+        await getToken(badLoginRequest);
+      } catch (e) {
+        expect(e.message).to.equal("Failed to get employee");
+        return;
+      }
+    });
+  });
+  describe("register", function () {
+    it("should return token", async () => {
+      const data = [
+        "eyJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE3MzYzNDEyOTAsImV4cCI6MTczNjM3MDA5MCwiZW1haWwiOiJUZXJyeUBnbWFpbC5jb20iLCJzdWIiOiJUZXJyeUBnbWFpbC5jb20iLCJpc3MiOiJLYWlub3MifQ.Dv1_YV2zS_I09SFQ-Ybach1V_6nLxJr3uLkEmF3TaHk",
+      ];
+
+      mock
+        .onPost("http://localhost:8080/api/auth/login", registerRequest)
+        .reply(200, data);
+      const results = await getToken(loginRequest);
+
+      expect(results).to.deep.equal(data);
+    });
+
+    it("should throw exception when 500 returned by axios", async () => {
+      mock
+        .onPost("http://localhost:8080/api/auth/login", badLoginRequest)
+        .reply(500);
+      try {
+        await getToken(loginRequest);
+      } catch (e) {
+        expect(e.message).to.equal("Failed to create employee");
+        return;
```

no try catches in tests, we want them to fail

---

We want to check what specific error is thrown, so we catch and then check the error message. 
I would argue tho that instead of returning it should it should re-throw

---

Actually no I re-read the code, id argue this is perfectly fine

If an error is thrown we want to check what the error says. If it's what we expected it will pass line 74 and will need to return without throwing.
If the error isn't what we expect, line 74 will raise a new exception and the test will fail